agent/state.go

```diff
@@ -63,6 +64,7 @@ func NewStateMachine(logger *slog.Logger) *StateMachine {
 
 	sm.Transitions[receivingAlgorithm] = make(map[event]state)
 	sm.Transitions[receivingAlgorithm][algorithmReceived] = receivingData
+	sm.Transitions[receivingAlgorithm][algoReceivedNoData] = running
```

Comment:
Put an `if` here and transition to data receiving or running state im dépendance of data length

Reply:
@rodneyosodo https://github.com/ultravioletrs/cocos/blob/d76074ae41b03c794f0dcb3da73a4ec0f67906ff/agent/state.go#L48-L77 change this to accept manifest, and define state transitions based on contents of manifest and as a result there will be no need to change service